core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/entity/player/BedrockActionTranslator.java

```diff
@@ -88,73 +102,74 @@ public void translate(GeyserSession session, PlayerActionPacket packet) {
 
                 // Needed here since 1.19.81 for dimension switching
                 session.getEntityCache().updateBossBars();
-                break;
-            case START_SWIMMING:
+            }
+            case START_SWIMMING -> {
                 if (!entity.getFlag(EntityFlag.SWIMMING)) {
                     ServerboundPlayerCommandPacket startSwimPacket = new ServerboundPlayerCommandPacket(entity.getEntityId(), PlayerState.START_SPRINTING);
                     session.sendDownstreamGamePacket(startSwimPacket);
 
                     session.setSwimming(true);
                 }
-                break;
-            case STOP_SWIMMING:
+            }
+            case STOP_SWIMMING -> {
                 // Prevent packet spam when Bedrock players are crawling near the edge of a block
                 if (!session.getCollisionManager().mustPlayerCrawlHere()) {
                     ServerboundPlayerCommandPacket stopSwimPacket = new ServerboundPlayerCommandPacket(entity.getEntityId(), PlayerState.STOP_SPRINTING);
                     session.sendDownstreamGamePacket(stopSwimPacket);
 
                     session.setSwimming(false);
                 }
-                break;
-            case START_GLIDE:
+            }
+            case START_GLIDE -> {
                 // Otherwise gliding will not work in creative
                 ServerboundPlayerAbilitiesPacket playerAbilitiesPacket = new ServerboundPlayerAbilitiesPacket(false);
                 session.sendDownstreamGamePacket(playerAbilitiesPacket);
-            case STOP_GLIDE:
+            }
+            case STOP_GLIDE -> {
```

Comment:
This is actually intended - we need to stop flying before being able to activate the elytra. However, this could probably just be copied to the start_flying case - the logic is probably due for a revision (e.g. guardian the flight disabling with a check whether the player is even flying)

Reply:
Fixed in latest commit